test/e2e/toolchaincluster_test.go

```diff
@@ -174,3 +176,86 @@ func caBundle(bundle string) clusterOption {
 		c.Spec.CABundle = bundle
 	}
 }
+
+func TestForceSynchronization(t *testing.T) {
+	// given
+	toolchainClusterList := &toolchainv1alpha1.ToolchainClusterList{}
+	ctx, hostAwait, memberAwait, member2Await := WaitForDeployments(t, toolchainClusterList)
+	defer ctx.Cleanup()
+	hostAwait.UpdateToolchainConfig(
+		testconfig.AutomaticApproval().Enabled(),
+		testconfig.Metrics().ForceSynchronization(false))
+	t.Cleanup(func() {
+		err := hostAwait.ScaleDeployment(hostAwait.Namespace, "host-operator", 1)
+		require.NoError(t, err)
+	})
+	// before creating a batch of users, let's remove all remainings of previous tests, so
+	// we know exactly what numbers we're dealing with
+	userSignups := &toolchainv1alpha1.UserSignupList{}
+	err := hostAwait.Client.List(context.TODO(), userSignups, client.InNamespace(hostAwait.Namespace))
+	require.NoError(t, err)
+	for _, userSignup := range userSignups.Items {
+		err := hostAwait.Client.Delete(context.TODO(), &userSignup)
+		require.NoError(t, err)
+	}
+	// then wait until all UserSignups have been deleted
+	for _, userSignup := range userSignups.Items {
+		hostAwait.WaitUntilUserSignupDeleted(userSignup.Name)
+	}
```

Comment:
This deletion of all usersignups logic seems like it could be useful as a common function. WDYT?

Reply:
yes, that's a good idea, let me do that

Reply:
done in 85154de